inc/c_util/tqueue_ll.h

```diff
@@ -378,13 +385,45 @@ TQUEUE_POP_RESULT TQUEUE_LL_POP(C)(TQUEUE_LL(T) tqueue, T* item, void* copy_item
     return result;                                                                                                                                                  \
 }                                                                                                                                                                   \
 
+/*introduces a function definition for tqueue_pop*/
+#define TQUEUE_LL_GET_COUNT_DEFINE(C, T)                                                                                                                            \
+int64_t TQUEUE_LL_GET_COUNT(C)(TQUEUE_LL(T) tqueue)                                                                                                                 \
+{                                                                                                                                                                   \
+    int64_t result;                                                                                                                                                 \
+    /* Codes_SRS_TQUEUE_22_001: [ If tqueue is NULL then TQUEUE_GET_COUNT(T) shall return zero. ]*/                                                                 \
+    if (tqueue == NULL)                                                                                                                                             \
+    {                                                                                                                                                               \
+        LogError("Invalid arguments: TQUEUE_LL(" MU_TOSTRING(T) ") tqueue=%p.", tqueue);                                                                            \
+        result = 0;                                                                                                                                                 \
+    }                                                                                                                                                               \
+    else                                                                                                                                                            \
+    {                                                                                                                                                               \
+        /* Codes_SRS_TQUEUE_22_002: [ TQUEUE_GET_COUNT(T) shall obtain the current head queue by calling interlocked_add_64. ]*/                                    \
+        int64_t current_head = interlocked_add_64((volatile_atomic int64_t*)&tqueue->head, 0);                                                                      \
```

Comment:
Should this be `interlocked_add_64(&tqueue->head, 0) % tqueue->queue_size`

Reply:
interlocked_add_64(&tqueue->head, 0) % tqueue->queue_size would get the index to the queue.  GET_COUNT is trying to get the difference between the tail and head.